pkg/reconciler/vertex/controller.go

```diff
@@ -228,6 +230,7 @@ func (r *vertexReconciler) reconcile(ctx context.Context, vertex *dfv1.Vertex) (
 			if err := r.client.Create(ctx, pod); err != nil {
 				log.Errorw("Failed to create pod", zap.String("pod", pod.Name), zap.Error(err))
 				vertex.Status.MarkPhaseFailed("CreatePodFailed", err.Error())
+				r.recorder.Event(pod, corev1.EventTypeWarning, "CreatePodFailed", err.Error())
```

Comment:
For vertex controller, I think for ALL occurrences of `vertex.Status.MarkPhaseFailed`, we can send a event.

It seems `log.Errorw`, `vertex.Status.MarkPhaseFailed`, `r.recorder.Event` and `return ctrl.Result{}, err` are a common sequence of operations, consider extract them into a single helper function.

Reply:
Helper function:
```
func (r *vertexReconciler) markPhaseLogEvent(vertex *dfv1.Vertex, log *zap.SugaredLogger, reason, message, logMsg string, logWith ...interface{}) {
	log.Errorw(logMsg, logWith)
	vertex.Status.MarkPhaseFailed(reason, message)
	r.recorder.Event(vertex, corev1.EventTypeWarning, reason, message)
}
```

Condensed logic vs with helper:
```
r.markPhaseLogEvent(vertex, log, "FindISBSvcFailed", err.Error(), "Failed to get ISB Service", zap.String("isbsvc", isbSvcName), zap.Error(err))
// log.Errorw("Failed to get ISB Service", zap.String("isbsvc", isbSvcName), zap.Error(err))
// vertex.Status.MarkPhaseFailed("FindISBSvcFailed", err.Error())
// r.recorder.Event(vertex, corev1.EventTypeWarning, "FindISBSvcFailed", err.Error())
```